qubes-rpc/qvm-copy-to-vm.kde

```diff
@@ -20,21 +20,35 @@
 #
 #
 
+if type kdialog 2> /dev/null; then
+kdlg=1
+else kdlg=0
+fi
+
+if [ $kdlg -eq 1 ] ; then
 VM=$(kdialog -inputbox "Enter the VM name to send files to:")
+else
+VM=$(zenity --entry --text="Enter the VM name to send files to:")
+fi
+
 if [ X$VM = X ] ; then exit 0 ; fi
 
 SIZE=$(du --apparent-size -c -- "$@" 2> /dev/null | tail -1 | cut -f 1)
+if [ $kdlg -eq 1 ] ;then
 REF=$(kdialog --progressbar "Copy progress")
+fi
 qdbus $REF org.freedesktop.DBus.Properties.Set "" maximum $SIZE
 
 export PROGRESS_TYPE=gui
 
 /usr/lib/qubes/qrexec-client-vm $VM qubes.Filecopy \
 	/usr/lib/qubes/qfile-agent "$@" |
+if [ $kdlg -eq 1 ] ;then
 (while read sentsize ; do
 	CURRSIZE=$(($sentsize/1024))
 	qdbus $REF  org.freedesktop.DBus.Properties.Set "" value $CURRSIZE
 done)
+fi
```

Comment:
Keep in mind the output of `qrexec-client-vm` above is piped here, so probably some `else` clause should be used. Otherwise `qfile-agent` will get SIGPIPE on reporting progress. Or simply set `PROGRESS_TYPE` to empty string.